core/classifiers/CodeClassifier/winnowing.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions for using winnowing as pre-processing strp."""
+
+
+def hash_generator(token_to_id, tokens):
+    """Generate hash for tokens in 'tokens' using 'token_to_id'.
+
+    Args:
+        token_to_id: dict. A dictionary which maps each token to a unique ID.
+        tokens: list(str). A list of tokens.
+
+    Returns:
+        int. Hash value generated for tokens in 'tokens' using 'token_to_id'.
+    """
+    hash_val = 0
+    n = len(tokens) - 1
+    for x in tokens:
+        hash_val += token_to_id[x] * (len(token_to_id) ** n)
```

Comment:
I don't know how much more processing ** takes compared to *, but if this might be a concern (especially since it's run in a for loop many many times) you might consider looking at something like Horner's method to keep track of the "current power of the base" instead.